Clean up getting Boolean values

taers232c opened the pull request:
Update getString to take minimum/maximum length
One more course id clean up at 2508

jay0lee commented on src/gam.py:
```diff
@@ -151,11 +151,32 @@ def csvFieldErrorExit(fieldName, fieldNames):
 def printLine(message):
   sys.stdout.write(message+u'\n')
 #
+def getBoolean(item, i):
```
Please don't pass i around like this, it's only going to make cleaning up our sys.argv usage much harder one day. Instead just pass the necessary sys.argv values as needed. That way calling the function doesn't depend on position of arguments.

taers232c commented:
Jay,

1) foo = getBoolean("xxx", i)
2) foo = getBoolean("xxx", sys.argv[i])

How is 2 less dependent on the position of the arguments?

1) Gives the user a much better message is case of an error


$ gam user testuser1 vacation

ERROR: Expected value on|yes|enabled|true|1 or off|no|disabled|false|0 for
gam <users> vacation

2) This tells the user nothing

$ gam user testuser1 vacation

GAM 4.32 - https://git.io/gam

Jay Lee <jay0lee@gmail.com>

Python 2.7.14 64-bit final

google-api-python-client 1.6.4

oauth2client 4.1.2

Darwin-15.6.0-x86_64-i386-64bit x86_64

Path: /Users/admin/Documents/GoogleApps/GAM


Usage: gam [OPTIONS]...


GAM. Retrieve or set G Suite domain,

user, group and alias settings. Exhaustive list of commands

can be found at: https://github.com/jay0lee/GAM/wiki


Examples:

gam info domain

gam create user jsmith firstname John lastname Smith password secretpass

gam update user jsmith suspended on

gam.exe update group announcements add member jsmith

...


Ross

On Mon, Oct 23, 2017 at 7:26 AM, Jay Lee <notifications@github.com> wrote:

> *@jay0lee* commented on this pull request.
> ------------------------------
>
> In src/gam.py
> <https://github.com/jay0lee/GAM/pull/625#discussion_r146282291>:
>
> > @@ -151,11 +151,32 @@ def csvFieldErrorExit(fieldName, fieldNames):
>  def printLine(message):
>    sys.stdout.write(message+u'\n')
>  #
> +def getBoolean(item, i):
>
> Please don't pass i around like this, it's only going to make cleaning up
> our sys.argv usage much harder one day. Instead just pass the necessary
> sys.argv values as needed. That way calling the function doesn't depend on
> position of arguments.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jay0lee/GAM/pull/625#pullrequestreview-71201064>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIU8LxtgCqc4f-FxgdUXFaGNs6kiAjRRks5svKITgaJpZM4QAB1w>
> .
>



-- 
Ross Scroggs
ross.scroggs@gmail.com

taers232c commented:
Jay,

I don't understand your concern. "That way calling the function doesn't
depend on position of arguments."

Old Style:
Fixed position:

def doVacation(users):

  if sys.argv[4].lower() in true_values:

    enable = True

  elif sys.argv[4].lower() in false_values:

    enable = False

  else:

    print u'ERROR: value for "gam <users> vacation" must be true or false;
got %s' % sys.argv[4]

    sys.exit(2)


Variable position:

    elif myarg == u'admin':

      if sys.argv[i+1].lower() in true_values:

        admin_body[u'status'] = True

      elif sys.argv[i+1].lower() in false_values:

        admin_body[u'status'] = False

      else:

        print u'ERROR: admin must be on or off; got %s' % sys.argv[i+1]

        sys.exit(2)

      i += 2



My Style:

Fixed position:

def doVacation(users):

  enable = getBoolean(u'gam <users> vacation', 4)


Variable position:

    elif myarg == u'admin':

      admin_body[u'status'] = getBoolean(myarg, i+1)

      i += 2


Your Suggested Style:
Fixed position:

def doVacation(users):

  enable = getBoolean(u'gam <users> vacation', sys.argv[4])

Variable position:

    elif myarg == u'admin':

      admin_body[u'status'] = getBoolean(myarg, sys.argv[i+1])

      i += 2

The calling function has always defined the order of the arguments for the
fixed position arguments as in doVacation; I'm not changing that.
If we want enable to be in sys.argv[5] instead of sys.argv[4] in
doVacation, everyone has to change 4 to 5.
For the variable position arguments, nothing has changed; myarg is in
sys.argv[i], the value is in sys.argv[i+1].
Passing i to the routine instead of sys.argv[i] let's you give the user a
better message when value is omitted.

Ross

On Mon, Oct 23, 2017 at 7:40 AM, Ross Scroggs <ross.scroggs@gmail.com>
wrote:

> Jay,
>
> 1) foo = getBoolean("xxx", i)
> 2) foo = getBoolean("xxx", sys.argv[i])
>
> How is 2 less dependent on the position of the arguments?
>
> 1) Gives the user a much better message is case of an error
>
>
> $ gam user testuser1 vacation
>
> ERROR: Expected value on|yes|enabled|true|1 or off|no|disabled|false|0 for
> gam <users> vacation
>
> 2) This tells the user nothing
>
> $ gam user testuser1 vacation
>
> GAM 4.32 - https://git.io/gam
>
> Jay Lee <jay0lee@gmail.com>
>
> Python 2.7.14 64-bit final
>
> google-api-python-client 1.6.4
>
> oauth2client 4.1.2
>
> Darwin-15.6.0-x86_64-i386-64bit x86_64
>
> Path: /Users/admin/Documents/GoogleApps/GAM
>
>
> Usage: gam [OPTIONS]...
>
>
> GAM. Retrieve or set G Suite domain,
>
> user, group and alias settings. Exhaustive list of commands
>
> can be found at: https://github.com/jay0lee/GAM/wiki
>
>
> Examples:
>
> gam info domain
>
> gam create user jsmith firstname John lastname Smith password secretpass
>
> gam update user jsmith suspended on
>
> gam.exe update group announcements add member jsmith
>
> ...
>
>
> Ross
>
> On Mon, Oct 23, 2017 at 7:26 AM, Jay Lee <notifications@github.com> wrote:
>
>> *@jay0lee* commented on this pull request.
>> ------------------------------
>>
>> In src/gam.py
>> <https://github.com/jay0lee/GAM/pull/625#discussion_r146282291>:
>>
>> > @@ -151,11 +151,32 @@ def csvFieldErrorExit(fieldName, fieldNames):
>>  def printLine(message):
>>    sys.stdout.write(message+u'\n')
>>  #
>> +def getBoolean(item, i):
>>
>> Please don't pass i around like this, it's only going to make cleaning up
>> our sys.argv usage much harder one day. Instead just pass the necessary
>> sys.argv values as needed. That way calling the function doesn't depend on
>> position of arguments.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jay0lee/GAM/pull/625#pullrequestreview-71201064>, or mute
>> the thread
>> <https://github.com/notifications/unsubscribe-auth/AIU8LxtgCqc4f-FxgdUXFaGNs6kiAjRRks5svKITgaJpZM4QAB1w>
>> .
>>
>
>
>
> --
> Ross Scroggs
> ross.scroggs@gmail.com
>



-- 
Ross Scroggs
ross.scroggs@gmail.com

taers232c commented:
Jay,

I don't understand your concern. "That way calling the function doesn't
depend on position of arguments."

Old Style:
Fixed position:
```
def doVacation(users):
  if sys.argv[4].lower() in true_values:
    enable = True
  elif sys.argv[4].lower() in false_values:
    enable = False
  else:
    print u'ERROR: value for "gam <users> vacation" must be true or false;
got %s' % sys.argv[4]
    sys.exit(2)
```
Variable position:

    elif myarg == u'admin':
      if sys.argv[i+1].lower() in true_values:
        admin_body[u'status'] = True
      elif sys.argv[i+1].lower() in false_values:
        admin_body[u'status'] = False
      else:
        print u'ERROR: admin must be on or off; got %s' % sys.argv[i+1]
        sys.exit(2)
      i += 2

My Style:
Fixed position:
```
def doVacation(users):
  enable = getBoolean(u'gam <users> vacation', 4)
```
Variable position:
```
    elif myarg == u'admin':
      admin_body[u'status'] = getBoolean(myarg, i+1)
      i += 2
```

Your Suggested Style:
Fixed position:
```
def doVacation(users):
  enable = getBoolean(u'gam <users> vacation', sys.argv[4])
```
Variable position:
```
    elif myarg == u'admin':
      admin_body[u'status'] = getBoolean(myarg, sys.argv[i+1])
      i += 2
```
The calling function has always defined the order of the arguments for the
fixed position arguments as in doVacation; I'm not changing that.
If we want enable to be in sys.argv[5] instead of sys.argv[4] in
doVacation, everyone has to change 4 to 5.
For the variable position arguments, nothing has changed; myarg is in
sys.argv[i], the value is in sys.argv[i+1].
Passing i to the routine instead of sys.argv[i] let's you give the user a
better message when value is omitted.

Ross

taers232c commented:
Jay,

Any thoughts on proceeding?

Ross

taers232c commented:
Jay,

After five weeks when no response to my comments, I'll close this out.

Ross
